src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -9243,7 +9249,7 @@ public void SkipAnalyzersFlagFiltersAnalyzers()
             var srcFile = Temp.CreateFile().WriteAllText(@"class C {}");
             var srcDirectory = Path.GetDirectoryName(srcFile.Path);
 
-            var args = new List<string>() { "/reportanalyzer", "/t:library", "/a:" + Assembly.GetExecutingAssembly().Location, srcFile.Path };
+            var args = new List<string>() { "/reportanalyzer", "/t:library", "/a:" + typeof(DoNothingGenerator).Assembly.Location, srcFile.Path };
```

Comment:
> I considered it but emitting to disk **add a large layer of complexity to the tests** (also some maintenance issues). Most of our tests aren't interested in the act of round trippping to disk, they just want to verify that such and such analyzer has such and such behavior. Simpler to just grab in memory and use. 
> 

I feel like **this** means we haven't exposed the right helpers for this purpose. 

> Indeed that is true. Our existing tests suffer from this problem. In general I don't think we should be using that switch broadly to load our test anaylzers. Yes great for a number of cases but I think in general we should prefer specifying the analyzers we want to use for a given test vs. indirectly passing an assembly. 
> 

It's definitely more precise to test at the layer where you pass the exact types of analyzers in use. But it's still necessary to test at the layer where analyzer discovery is performed and it feels like existing tests shouldn't change incidentally because a new analyzer was needed for a new test. I feel that layer tends to be under-tested in part due to the difficulty of setting up the inputs to it.

All that said, I don't see a blocking need to change the way we obtain the analyzer here for this PR.